ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/common/CoreEnvironmentBean.java

```diff
@@ -105,6 +106,25 @@ public void setNodeIdentifier(String nodeIdentifier) throws CoreEnvironmentBeanE
         }
 
     	this.nodeIdentifier = nodeIdentifier;
+        this.nodeIdentifierBytes = nodeIdentifier.getBytes(StandardCharsets.UTF_8);
+    }
+
+    public byte[] getNodeIdentifierBytes() {
+        return this.nodeIdentifierBytes;
+    }
+
+    public void setNodeIdentifier(byte[] bytes) throws CoreEnvironmentBeanException
```

Comment:
Wouldn't make sense to use this method as the default one? To be more specific, from `public void setNodeIdentifier(String nodeIdentifier)`, we could invoke directly the new `public void setNodeIdentifier(byte[] bytes)`. Moreover, it would be great if we can specify that `public void setNodeIdentifier(String nodeIdentifier)` uses UTF8 in its javadoc. WDYT @mmusgrov?

Reply:
I know that you see two lines repeated in each method but I want to minimise the change set. The goal of the PR is to give the caller the option of explicitly setting the bytes for the purpose of encoding Xids in order to avoid the problems caused by variable width encoding. On the other point, I don't want to force them to use a specific a charset because that is just an encoding and they can decode it in any character set that suits them.

Reply:
I see your point :-)